app/models/manageiq/providers/hawkular/datawarehouse_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,105 @@
+class ManageIQ::Providers::Hawkular::DatawarehouseManager::EventCatcher::Runner <
```

Comment:
> If we try to do both here I fear this would not get merged.

absolutely. But I think it will even increase your merge chance if you 

* refactor middleware to use the mixin (this should be mostly a code movement PR)
* use the mixin here

This first PR should get in fast and this one get smaller :)
Plus you make sure the extraction actually works for middleware and dwh
Plus we dont risk stale code duplication (in case nobody does the middleware refactoring)